src/Evolution/Systems/CurvedScalarWave/BoundaryConditions/ConstraintPreservingSphericalRadiation.hpp

```diff
@@ -0,0 +1,168 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <memory>
+#include <optional>
+#include <pup.h>
+#include <string>
+#include <type_traits>
+
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Evolution/BoundaryConditions/Type.hpp"
+#include "Evolution/Systems/CurvedScalarWave/BoundaryConditions/BoundaryCondition.hpp"
+#include "Evolution/Systems/CurvedScalarWave/Tags.hpp"
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "PointwiseFunctions/AnalyticData/Tags.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/AnalyticSolution.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Time/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace domain::Tags {
+template <size_t Dim, typename Frame>
+struct Coordinates;
+}  // namespace domain::Tags
+/// \endcond
+
+namespace CurvedScalarWave::BoundaryConditions {
+
+/*!
+ * \brief Implements constraint-preserving boundary conditions with a second
+ * order Bayliss-Turkel radiation boundary condition.
+ *
+ * \details The Bayliss-Turkel boundary conditions are technically only valid in
+ * flat space and should therefore only be used at boundaries where the
+ * background spacetime is approximately Minkwoski such as (sufficiently far
+ * out) outer boundaries for asymptotically flat spacetimes. Small reflections
+ * are still likely to occur.
+ *
+ * The constraint-preserving part of the boundary conditions are set on the time
+ * derivatives of the evolved fields according to \cite Holst2004wt . The
+ * physical Bayliss-Turkel boundary conditions are additionally set onto the
+ * time derivative of \f$\Pi\f$.
+ *
+ * The constraints are defined as follows:
+ *
+ * \f{align*}{
+ *  \mathcal{C}_i&=\partial_i\Psi - \Phi_i=0, \\
+ *  \mathcal{C}_{ij}&=\partial_{[i}\Phi_{j]}=0
+ * \f}
+ *
+ * The boundary conditions are then given by:
```

Comment:
Do these equations follow from Holst2004wt, Sec. III B, and if so, how? Could you expand on that a bit in the docs?

Reply:
I expanded this step a bit.